cmd/sippy/syncvariants.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"os"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/openshift/sippy/pkg/variantregistry"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"google.golang.org/api/option"
+
+	"github.com/openshift/sippy/pkg/flags"
+)
+
+type SyncJobVariantsFlags struct {
+	BigQueryFlags    *flags.BigQueryFlags
+	GoogleCloudFlags *flags.GoogleCloudFlags
+}
+
+func NewSyncJobVariantsFlags() *SyncJobVariantsFlags {
+	return &SyncJobVariantsFlags{
+		BigQueryFlags:    flags.NewBigQueryFlags(),
+		GoogleCloudFlags: flags.NewGoogleCloudFlags(),
+	}
+}
+
+func (f *SyncJobVariantsFlags) BindFlags(fs *pflag.FlagSet) {
+	f.BigQueryFlags.BindFlags(fs)
+	f.GoogleCloudFlags.BindFlags(fs)
+}
+
+func NewSyncJobVariantsCommand() *cobra.Command {
```

Comment:
One thing that crossed my mind is that this is basically a loader, I wonder if the variant registry shouldn't just be a dataloader in pkg/dataloader.

Then there'd be no need for another command, you'd just run `./sippy load --loader variants`

I wouldn't hold up merging on this, but you do get some benefits like this gets merged up to metrics on grafana, and you get plotting of how long it takes, error counts, etc.

Reply:
So I have two new commands in this pr, load variants, which generates a json file, and sync variants, which syncs it to the db and takes an input filename param. I think sync is the better fit for load command, but input file is an issue. Two commands as the latter was intended for third party users, whereas the current "load variants" command was just for our logic, and maybe QEs.

I think I'll rename my load-job-variants to generate-job-variants to output the json file, and then I can move sync over to be a loader, and that will use an assumed filename? I can't really add a --input param as load takes multiple loaders and we wouldn't know which it was intended for. Seem reasonable?

Reply:
>  then I can move sync over to be a loader, and that will use an assumed filename? I can't really add a --input param as load takes multiple loaders and we wouldn't know which it was intended for. Seem reasonable?

Ah I didn't think about that - it's probably OK to keep this as you have it then. If you want the data loader interface, I don't think it's a problem to add a parameter that's only used for one loader.  Not all loaders use all all arguments